packages/bundler-plugin-core/src/index.ts

```diff
@@ -510,6 +532,59 @@ export function createRollupDebugIdUploadHooks(
   };
 }
 
+export function createReactAnnotateHooks() {
+  return {
+    async transform(code: string, id: string): Promise<TransformResult> {
+      // id may contain query and hash which will trip up our file extension logic below
+      const idWithoutQueryAndHash = stripQueryAndHashFromPath(id);
+
+      if (idWithoutQueryAndHash.match(/\\node_modules\\|\/node_modules\//)) {
+        return null;
+      }
+
+      // We will only apply this plugin on jsx and tsx files
+      if (![".jsx", ".tsx"].some((ending) => idWithoutQueryAndHash.endsWith(ending))) {
+        return null;
+      }
+
+      const parserPlugins = [];
+      if (idWithoutQueryAndHash.endsWith(".jsx")) {
+        parserPlugins.push("jsx");
+      } else if (idWithoutQueryAndHash.endsWith(".tsx")) {
+        parserPlugins.push("jsx");
+        parserPlugins.push("typescript");
+      }
+
+      try {
+        const result = await transformAsync(code, {
+          plugins: [[reactAnnotate]],
```

Comment:
We need to pass in the `filename` to the options which is passed to as an argument to the `transform` hook, so we may not be able to do this because we will still have to reconstruct the object with the constants